pkg/getty/etc/profile.d/namespace.sh

```diff
@@ -1 +1,23 @@
-export PS1="(ns: getty) $PS1"
+#/bin/sh
+
+in_a_ns() {
+	IN_A_NS=0
+	NOT_IN_A_NS=1
+
+	# If PID 1 != /sbin/init then we are in a container
+	if ! pgrep /sbin/init > /dev/null; then
+		return $IN_A_NS
+	fi
+
+	# If containerd or runc are in the pstree then we're probably in a namespace
+	if pstree | grep -qE "(containerd|runc)"; then
+		return $IN_A_NS
+	fi
```

Comment:
Be careful that your `grep` isn't matching itself:
```
(ns: getty) linuxkit-525400123456:~# ps aux| grep -E containerd\|runc
  391 root       0:00 /usr/bin/containerd
  413 root       0:00 containerd-shim --namespace default --address /run/containerd/containerd.sock
  465 root       0:00 containerd-shim --namespace default --address /run/containerd/containerd.sock
  507 root       0:00 containerd-shim --namespace default --address /run/containerd/containerd.sock
  557 root       0:00 grep -E containerd|runc
```

I usually use a single character group to defeat the unwanted matching, but you need it in all the alternatives i.e.

```
(ns: getty) linuxkit-525400123456:~# ps aux| grep -E [c]ontainerd\|[r]unc
  391 root       0:00 /usr/bin/containerd
  413 root       0:00 containerd-shim --namespace default --address /run/containerd/containerd.sock
  465 root       0:00 containerd-shim --namespace default --address /run/containerd/containerd.sock
  507 root       0:00 containerd-shim --namespace default --address /run/containerd/containerd.sock
```